### PR TITLE
fix: build for any kind of arm combination

### PR DESCRIPTION
**Pull request by ayounes-nviso**
* This simplification allows to use src/CMakeLists.txt directly
  within a project that has its own toolchain.
* Introduce NCNN_CMAKE_VERBOSE option to reduce cmake verbosity

### CODE REVIEW

**Comment by tencent-adm** [comment]
[CLA assistant check](https://cla.opensource.tencent.com/Tencent/ncnn?pullRequest=657) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.tencent.com/Tencent/ncnn?pullRequest=657) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.tencent.com/check/Tencent/ncnn?pullRequest=657) it.</sub>

**Comment by ayounes-nviso** [comment]
Hello,
I am trying to sign the CLA but it always times out...

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -31,44 +31,37 @@ macro(ncnn_add_layer class)
...
-            OR (UNIX AND ("${CMAKE_SYSTEM_PROCESSOR}" STREQUAL "armv7l"))
-            OR (UNIX AND ("${CMAKE_SYSTEM_PROCESSOR}" STREQUAL "aarch64"))
-            OR (IOS AND ("${CMAKE_OSX_ARCHITECTURES}" STREQUAL "armv7"))
-            OR (IOS AND ("${CMAKE_OSX_ARCHITECTURES}" STREQUAL "arm64"))
-            OR (IOS AND ("${CMAKE_OSX_ARCHITECTURES}" STREQUAL "armv7;arm64")))
-            if(EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/layer/arm/${name}_arm.cpp")
-                list(APPEND ncnn_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/layer/arm/${name}_arm.cpp")
-                set(WITH_LAYER_${name}_arm 1)
-            endif()
+        if(CMAKE_SYSTEM_PROCESSOR MATCHES "^(arm|aarch64)")
```
[the remark this answers is not in the document]
Comment (ayounes-nviso) [reply src/CMakeLists.txt]: The initial code does not do something special for osx or ios either. As of now you only have specialisation for ARM or for Intel.
If for example you envision to have specific optimisation per arch and per os then the same logic can be applied:

if(IOS)
    set(os IOS)
elseif(ANDROID)
    set(os android)
end()

And then you can use ${os} in your os specific layer filename.

Hope that answers the question :-)

[the remark this answers is not in the document]
Comment (nihui) [reply src/CMakeLists.txt]: when building for ios, CMAKE_OSX_ARCHITECTURES var holds the target arch info but not CMAKE_SYSTEM_PROCESSOR, which may report as the host x86_64 still.
in addition, the IOS platform may refer the armv7/arm64 iPhoneOS, or x86_64 iPhoneSimulator. so as for android platform. platform logic is not suitable here.

[the remark this answers is not in the document]
Comment (ayounes-nviso) [reply src/CMakeLists.txt]: You are completely right of course!
Sorry, for some reason I didn't see that the code used  CMAKE_OSX_ARCHITECTURES at the first place.

Will update the PR now.

Thanks,
Abdel

**Comment by nihui** [comment]
Thanks for your contribution !

**Pull request merged**